includes/reader-revenue/my-account/class-woocommerce-my-account.php

```diff
@@ -131,16 +150,67 @@ public static function handle_password_reset_request() {
 		}
 	}
 
+	/**
+	 * Handle magic link request.
+	 */
+	public static function handle_magic_link_request() {
+		if ( ! is_user_logged_in() ) {
+			return;
+		}
+		$nonce = filter_input( INPUT_GET, self::SEND_MAGIC_LINK_PARAM, FILTER_SANITIZE_STRING );
+
+		if ( $nonce ) {
+			$is_error = false;
+			if ( wp_verify_nonce( $nonce, self::SEND_MAGIC_LINK_PARAM ) ) {
+				$result  = Reader_Activation::send_verification_email( wp_get_current_user() );
+				$message = __( 'Please check your email inbox for a link to verify your account.', 'newspack' );
+				if ( is_wp_error( $result ) ) {
+					Logger::log( 'Error sending verification email: ' . $result->get_error_message() );
+					$message  = __( 'Something went wrong.', 'newspack' );
+					$is_error = true;
+				}
+			} else {
+				$message  = __( 'Something went wrong.', 'newspack' );
+				$is_error = true;
+			}
+			wp_safe_redirect(
+				add_query_arg(
+					[
+						'message'  => $message,
+						'is_error' => $is_error,
+					],
+					remove_query_arg( self::SEND_MAGIC_LINK_PARAM )
+				)
+			);
+			exit;
+		}
+	}
+
+	/**
+	 * Check if the user is logged in and verified.
+	 */
+	public static function is_reader_verified() {
```

Comment:
```suggestion
	public static function is_user_verified() {
```

Considering we're checking RAS availability, I think the broader term is suitable here.

Reply:
Updated in d6f8e41